integration-tests/testsetups/ocr.go

```diff
@@ -824,70 +823,105 @@ func (o *OCRSoakTest) setFilterQuery() {
 		Msg("Filter Query Set")
 }
 
-// observeOCREvents subscribes to OCR events and logs them to the test logger
-// WARNING: Should only be used for observation and logging. This is not a reliable way to collect events.
-func (o *OCRSoakTest) observeOCREvents() error {
-	eventLogs := make(chan types.Log)
-	ctx, cancel := context.WithTimeout(testcontext.Get(o.t), 5*time.Second)
-	eventSub, err := o.seth.Client.SubscribeFilterLogs(ctx, o.filterQuery, eventLogs)
-	cancel()
-	if err != nil {
-		return err
-	}
+// pollingOCREvents Polls the blocks for OCR events and logs them to the test logger
+func (o *OCRSoakTest) pollingOCREvents(endTest <-chan time.Time) error {
+
+	// Keep track of the last processed block number
+	processedBlockNum := o.startingBlockNum - 1
 
 	go func() {
+		// TODO: Make this configurable
```

Comment:
For this PR or follow-up? Is there a ticket to link?

Reply:
Intention was for a follow up. Just created SHIP-3973 to cover it.